### ISSUE TITLE
MVP ToDo

### ISSUE THREAD

**Issue by gaurav-naik** [mntechnique/kisanmitra_erpnext#2]
Checklist from past discussions:
- [ ] Suggest possible improvements in custom fields on Issue doctype.
- [ ] Desk Access with restrictions for Government Officials
- [x] Map State, District, Division, Mandal, Revenue Village in Territory Tree
- [x] Script for migrating Contacts, Phone Calls, Comments and Cases
- [ ] Restrict issues Internal / External via permissions
- [ ] A Sms should be sent in Telugu when a case is created to the caller with Case.No
- [ ] A Sms sent to a user when he is assigned to case(Currently the system is built for sending Emails)

- [ ] Setup a server environment and set things on Digital Ocean to migrate the code base for Exotel Integration as well as other changes made for the customisation towards KisanMitra

After discussion at Hyderabad
- [ ] Fixtures export for custom field and other doctypes, howto
- [ ] Bench import howto
- [ ] Query reports as requested
- [ ] Weather data separate data processing project 

Additional Suggestions (including points discussed in Hyderabad) [WIP]
1. Treat all leads as Organizations by default,

**Comment by nanoharsha** [comment]
While importing data from the Contacts, you can omit the fields below : 

Contacts Last Contacted   Via | 
Contacts Created Time | 
Contacts Modified Time | 
Contacts Assigned To

**Comment by nanoharsha** [comment]
@gaurav-naik do find a list of Fields which are mandatory, and fields which need to be imported into ERPNext & wherever linked to other doctypes.

[List_Fields_KisanMItra_Import_MNtech.xlsx](https://github.com/mntechnique/kisanmitra_erpnext/files/1815696/List_Fields_KisanMItra_Import_MNtech.xlsx)

**Comment by saurabh-bhosale** [comment]
@nanoharsha Thanks! This list of mandatory fields helped. Also, would be good know the answers for below questions that are related to provided data dump.
@gaurav-naik Have made this list of doubts/issues that were encountered during data-mapping in import scripts. 

Points to be discussed:
1. What to do with Leads that have no contact number(important for import scripts as well as call popup in telephony)
2. Few leads in Lead Record have digits as their _contact last name_ field. Need to know if this has some purpose behind it or else we have planned to have a code that skips adding these leads' last name in Contact DocType which have digits in them. 
	For eg: Contact First Name-some_name, Contact Last Name-0,
3. How to handle random names in Lead Data( might result in bad information in telephony dashboards)
	For eg: Contact First Name-far, Contact Last Name-7,
			Contact First Name-Far, Contact Last Name-0,
			Contact First Name-Unknown, Contact Last Name-0,
			Contact First Name-........, Contact Last Name-MISSED,							
4. @nanoharsha What is the meaning of digits/numbers in _Handled By_ field of Phone Calls dump? Is it some user id for Users on your side?

**Comment by nanoharsha** [comment]
1. Majority of Contacts in Kisanmitra without contact numbers would be of Users who don't have mobile number and would have called from other phones, when we checked now we found a few duplicates also one with mobile number, one without, we would need to go through case by case and import. Can we import all for now, and then update the records again based on our analysis ?
2. The dates were kept to store as reminder on what day the farmer called(strange :)) , these were callers whose calls we missed and tried to follow up but were not reachable. We maintain a seperate Excel for calls we were unable to connect, so that data will be there too. But if we can have a mechanism here itself would be helpful, but we can tag all such cases as missed call farmers.
3. Where ever such data is filled, these are all are created with some dummy name for farmers as a primary requirement to make a call from Vtiger is to create a contact and hence team has created contacts with dummy data. IF they are still having dummy data, then these are farmers we were unable to get in touch with again, and would need to follow up after a certain period of time. So its extension to above point. 
4. As I have seen its a user id for each user created in Exotel, but we are unable to get it from the system , but if remove duplicates between cominatin of user name & id(digits/numbers), then we will see that there are only few uses and have unique combination. Some times we have renamed the same user for use by another person, hence you see multile ids created with same email id but has different user id in Exotel. I have shared a list of uses from Vtiger to @gaurav-naik
